### PR TITLE
[Doris On ES][Docs] refator documentation for doe

### PR DESCRIPTION
**Pull request by wuyunfeng**
The past few months Doris On ES make a lots of performance optimization and bug-fix,  we should supplement and reorganize Doris On ES's documentation

### CODE REVIEW

File: docs/zh-CN/extending-doris/doris-on-es.md [hunk 1]
```diff
@@ -30,72 +30,320 @@ Doris-On-ES将Doris的分布式查询规划能力和ES(Elasticsearch)的全文
...
-* scroll：ES内置的数据集游标特性，用来对数据进行流式扫描和过滤。
+### Doirs相关
+* FE：Frontend，Doris 的前端节点,负责元数据管理和请求接入
+* BE：Backend，Doris 的后端节点,负责查询执行和数据存储
 
+### ES相关
+* DataNode：ES的数据存储与计算节点
+* MasterNode：ES的Master节点，管理元数据、节点、数据分布等
+* scroll：ES内置的数据集游标特性，用来对数据进行流式扫描和过滤
+* _source: 导入时传入的原始JSON格式文档内容
```
Comment (EmmyMiao87) [inline docs/zh-CN/extending-doris/doris-on-es.md:46]: 用``将变量名括起来，可以避免下划线造成的斜体影响

File: docs/zh-CN/extending-doris/doris-on-es.md [hunk 2]
```diff
@@ -30,72 +30,320 @@ Doris-On-ES将Doris的分布式查询规划能力和ES(Elasticsearch)的全文
...
+* doc_values: ES/Lucene 中字段的列式存储定义
```
Comment (EmmyMiao87) [inline docs/zh-CN/extending-doris/doris-on-es.md:47]: 同上

File: docs/zh-CN/extending-doris/doris-on-es.md (hunk 1 above)
Comment (wuyunfeng) [reply docs/zh-CN/extending-doris/doris-on-es.md:46]: 这应该显示问题吧，表格里面的加\就可以了，类别的好像不用

File: docs/en/extending-doris/doris-on-es.md [hunk 3]
```diff
@@ -26,62 +26,314 @@ under the License.
...
+"type" = "doc",
+
+"user" = "root",
+"password" = "root"
+);
+```
+
+The following parameters are accepted by ES table:
+
+参数 | 说明
```
Comment (wutiangan) [inline docs/en/extending-doris/doris-on-es.md:130]: change it to english

File: docs/en/extending-doris/doris-on-es.md [hunk 4]
```diff
@@ -26,62 +26,314 @@ under the License.
...
+"type" = "doc",
+"user" = "root",
+"password" = "root",
+
+"enable_keyword_sniff" = "true"
+);
+```
+
+参数说明：
+
```
Comment (wutiangan) [inline docs/en/extending-doris/doris-on-es.md:241]: change it to english

File: docs/zh-CN/extending-doris/doris-on-es.md [hunk 5]
```diff
@@ -30,72 +30,320 @@ Doris-On-ES将Doris的分布式查询规划能力和ES(Elasticsearch)的全文
...
+参数 | 说明
+---|---
+**hosts** | ES集群地址，可以是一个或多个，也可以是ES前端的负载均衡地址
+**index** | 对应的ES的index名字
+**type** | index的type，不指定的情况会使用_doc
+**user** | ES集群用户名
+**password** | 对应用户的密码信息
+
+* ES 7.x之前的集群请注意在建表的时候选择正确的**索引类型type**
+* 认证方式仅支持Http Bastic认证，完全开源的ES集群用户和密码不需要指定，需要确保该用户有访问: /\_cluster/state/、\_nodes/http等路径权限和对index的读权限
```
Comment (wutiangan) [inline docs/zh-CN/extending-doris/doris-on-es.md:137]: "完全开源的ES集群用户和密码不需要指定"
This sentence doesn't read smoothly

File: docs/zh-CN/extending-doris/doris-on-es.md [hunk 6]
```diff
@@ -30,72 +30,320 @@ Doris-On-ES将Doris的分布式查询规划能力和ES(Elasticsearch)的全文
...
+smallint | &radic; | &radic; |  | |   |   |   |   |  
+int | &radic; |  &radic; | &radic; | |   |   |   |   |  
+bigint | &radic;  | &radic;  | &radic;  | &radic; |   |   |   |   | 
+float |   |   |   |   | &radic; |   |   |   |  
+double |   |   |   |   |   | &radic; |   |   |  
+char |   |   |   |   |   |   | &radic; | &radic; |  
+varchar |  |   |   |   |   |   | &radic; | &radic; |   
+date |   |   |   |   |   |   |   |   | &radic;|  
+datetime |   |   |   |   |   |   |   |   | &radic;|  
+
```
Comment (wutiangan) [inline docs/zh-CN/extending-doris/doris-on-es.md:173]: es support decimal？

File: docs/zh-CN/extending-doris/doris-on-es.md [hunk 7]
```diff
@@ -30,72 +30,320 @@ Doris-On-ES将Doris的分布式查询规划能力和ES(Elasticsearch)的全文
...
+#### 基本查询
 
 ```
 select * from es_table where k1 > 1000 and k3 ='term' or k4 like 'fu*z_'
 ```
 
-#### 扩展的esquery sql语法
+#### 扩展的esquery
 通过`esquery`函数将一些无法用sql表述的ES query如match、geoshape等下推给ES进行过滤处理，`esquery`的第一个列名参数用于关联`index`，第二个参数是ES的基本`Query DSL`的json表述，使用花括号`{}`包含，json的`root key`有且只能有一个，如match、geo_shape、bool等
```
Comment (wutiangan) [inline docs/zh-CN/extending-doris/doris-on-es.md:339]: query后面需要加个括号

File: docs/zh-CN/extending-doris/doris-on-es.md [hunk 8]
```diff
@@ -195,37 +443,30 @@ select * from es_table where esquery(k4, ' {
...
 
     是，比如_count相关的query等，ES内部会直接读取满足条件的文档个数相关的元数据，不需要对真实的数据进行过滤
+
+4. 聚合操作是否可以下推
+
+   目前Doris On ES不支持聚合操作如sum, avg, min/max 等下推，计算方式是批量流式的从ES获取所有满足条件的文档，然后在Doris中进行计算
+   
+5. 日期类型字段的过滤条件无法下推
+   
+   日期类型的字段因为时间格式的问题，大多数情况下都不不会下推；对于日期类型的过滤可以采用字符串形式，日期格式需要和ES中保持完全一致
```
Comment (wutiangan) [inline docs/zh-CN/extending-doris/doris-on-es.md:472]: 不不=不

File: docs/zh-CN/extending-doris/doris-on-es.md (hunk 6 above)
Comment (wuyunfeng) [reply docs/zh-CN/extending-doris/doris-on-es.md:173]: emm, not support decimal

**Review by wutiangan — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**